service_change.go

```diff
@@ -30,15 +37,62 @@ func (w *ServiceChangeWaiter) RefreshFunc() resource.StateRefreshFunc {
 			return nil, "", err
 		}
 
-		return service, service.State, nil
+		// If this is a Kafka service, wait for its component parts (i.e. Kafka Connect,
+		// Kafka REST, and Schema Registry) to be ready
+		state := service.State
+		if state == aivenTargetState && service.Type == "kafka" {
```

Comment:
@jelmersnoeck I'm not against this but this seems to tie into Issue #3 Feel like if we want this kind of behaviour that it should live inside a Kafka specific resource and this service resource has limited functionality but works with everything that is not more fleshed out in the terraform provider.

Any thoughts? cc: @briancharous && @benjigoldberg

Reply:
+1 on making a Kafka specific resource, but I am not opposed in having this logic in the general resource either.

I would suggest taking out the conditional here and putting all of that in the `kafkaServiceReady` function. In that function, we can do a conditional to validate if this is a Kafka resource, if not, we return true.

Reply:
Yeah, I agree with this. I think we'll make a follow up PR possibly next week to make this change, but I wanted to get this fix in first.